### DISCUSSION TITLE
Using Docker api from a function

### DISCUSSION THREAD

**Discussion by marcbenedi** [appwrite/appwrite#1451] [Q&A] [answered]
Greetings,

I was trying to create a function that can communicate with Docker: 
```python
import docker

client = docker.DockerClient(base_url='unix://var/run/docker.sock')

a = client.containers.list(all=True)
print(a)
```

When executing it from Appwrite I get the following error:
```
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```

I thought it would work because in the `docker-compose.yaml` there is the service `appwrite-worker-functions` which has access to `/var/run/docker.sock`.

Is there any way to configure the docker container that runs the function so I can add `/var/run/docker.sock` as a volume?
Otherwise, would you have any alternative to do that?

Thank you 🙏!

**Comment by eldadfux** [comment, marked as answer]
`appwrite-worker-functions` has access to the docker.sock so it can spin up **isolated** runtime environments to run your functions. Exposing the Docker API to any function will be considered a big security weakness.

If you still want to achieve that you can run your own Docker container with access to the docker.sock. This can container can expose an API that your Appwrite functions can use. Be careful and make sure only authorized clients can access this kind of proxy.

**Reply by marcbenedi** [reply]
Thank you for your reply.

I see. Then I will follow your suggestion and create an API that handles the communication between the functions and Docker.
